File: daily/python/791_custom_sort_string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        char_cnt = {char: 0 for char in order}
        for char in s:
            if char in char_cnt:
                char_cnt[char] += 1
        
        result = ""
        for char in order:
            result += char * char_cnt[char]
        
        for char in s:
            if char not in order:
                result += char
        return result

File: daily/cpp/791_custom_sort_string.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  string customSortString(string order, string s) {
    unordered_map<char, int> charCnt;
    for (char& c : order) {
      charCnt[c] = 0;
    }
    for (char& c : s) {
      charCnt[c]++;
    }

    string res;
    for (char& c : order) {
      res.append(charCnt[c], c);
    }
    for (char& c : s) {
      if (charCnt.find(c) == charCnt.end()) {
        res.push_back(c);
      }
    }
    return res;
  }
};